app/Http/Controllers/Rpc/WalletController.php

```diff
@@ -38,19 +40,24 @@ class WalletController extends Controller
 
     public function calculateWithdrawal(Request $request)
     {
+        $addressFrom = $this->getAdServerAdsAddress();
+        if (!AdsValidator::isAccountAddressValid($addressFrom)) {
+            Log::error("Invalid ADS address is set: ${addressFrom}");
+            return self::json([], Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+
         Validator::make($request->all(), [
             self::FIELD_AMOUNT => [self::VALIDATOR_RULE_REQUIRED, 'integer', 'min:1'],
             self::FIELD_TO => self::VALIDATOR_RULE_REQUIRED,
         ])->validate();
         $amount = $request->input(self::FIELD_AMOUNT);
         $addressTo = $request->input(self::FIELD_TO);
 
-        $addressFrom = $this->getAdserverAdsAddress();
-        if (!AdsValidator::isAccountAddressValid($addressFrom)
-            || !AdsValidator::isAccountAddressValid($addressTo)) {
+        if (!AdsValidator::isAccountAddressValid($addressTo)) {
             // invalid input for calculating fee
-            return self::json([], 422);
+            return self::json(['error' => 'invalid address'], Response::HTTP_UNPROCESSABLE_ENTITY);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "error" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)